template/registry_number.go

```diff
@@ -21,11 +22,21 @@ func parseNumberOperand(operand any) (float64, error) {
 		return 0, fmt.Errorf(format+": %w", append(args, mf2.ErrBadOperand)...)
 	}
 
+	var (
+		v   float64
+		err error
+	)
+
 	if operand == nil {
 		return errorf("operand is required")
 	}
 
-	v, err := castAs[float64](operand)
+	if s, ok := operand.(string); ok {
```

Comment:
does it make sense to use type switch here?

Reply:
It's possible to cast string to []byte, not sure about other types

Reply:
```go
switch v := operand.(type) {
default:
   // castAs
   // return err if errored
case nil:
  // return err
case string:
  // json.Unmarshal
  // return err if errored
}
```